pkg/kubectl/rolling_updater_test.go

```diff
@@ -132,165 +132,292 @@ func newRc(replicas int, desired int) *api.ReplicationController {
+Scaling foo-v2 up from 0 to 10 (by 2 per interval), scaling foo-v1 down from 10 to 0 (by 2 per interval)
+Updating foo-v1 replicas: 8, foo-v2 replicas: 2
+Performing acceptance check of foo-v2
+Updating foo-v1 replicas: 6, foo-v2 replicas: 4
+Performing acceptance check of foo-v2
+Updating foo-v1 replicas: 4, foo-v2 replicas: 6
+Performing acceptance check of foo-v2
+Updating foo-v1 replicas: 2, foo-v2 replicas: 8
+Performing acceptance check of foo-v2
+Updating foo-v1 replicas: 0, foo-v2 replicas: 10
+Performing acceptance check of foo-v2
+Update succeeded. Deleting foo-v1
+`,
+		}, {
+			oldRc(2), newRc(6, 6), Accepted, Percent(30),
+			[]fakeResponse{
+				// no existing newRc
+				{nil, fmt.Errorf("not found")},
+				// scaling iteration
+				{newRc(2, 6), nil},
+				{oldRc(1), nil},
+				// scaling iteration
+				{newRc(4, 6), nil},
+				{oldRc(0), nil},
+				// final scale up
+				{newRc(6, 6), nil},
+				// cleanup annotations
+				{newRc(6, 6), nil},
+				{newRc(6, 6), nil},
+				{newRc(6, 6), nil},
+			},
+			`Creating foo-v2
+Scaling foo-v2 up from 0 to 6 (by 2 per interval), scaling foo-v1 down from 2 to 0 (by 1 per interval)
+Updating foo-v1 replicas: 1, foo-v2 replicas: 2
+Performing acceptance check of foo-v2
+Updating foo-v1 replicas: 0, foo-v2 replicas: 4
+Performing acceptance check of foo-v2
+Scaling foo-v2 replicas: 4 -> 6
+Update succeeded. Deleting foo-v1
+`,
+		}, {
+			oldRc(10), newRc(3, 3), Accepted, Percent(30),
+			[]fakeResponse{
+				// no existing newRc
+				{nil, fmt.Errorf("not found")},
+				// scaling iteration
+				{newRc(1, 3), nil},
+				{oldRc(7), nil},
+				// scaling iteration
+				{newRc(2, 3), nil},
+				{oldRc(4), nil},
+				// scaling iteration
+				{newRc(3, 3), nil},
+				{oldRc(1), nil},
+				// final scale down
+				{oldRc(0), nil},
+				// cleanup annotations
+				{newRc(3, 3), nil},
+				{newRc(3, 3), nil},
+				{newRc(3, 3), nil},
+			},
+			`Creating foo-v2
+Scaling foo-v2 up from 0 to 3 (by 1 per interval), scaling foo-v1 down from 10 to 0 (by 3 per interval)
+Updating foo-v1 replicas: 7, foo-v2 replicas: 1
+Performing acceptance check of foo-v2
+Updating foo-v1 replicas: 4, foo-v2 replicas: 2
+Performing acceptance check of foo-v2
+Updating foo-v1 replicas: 1, foo-v2 replicas: 3
+Performing acceptance check of foo-v2
+Stopping foo-v1 replicas: 1 -> 0
+Update succeeded. Deleting foo-v1
+`,
 		},
 	}
 
 	for _, test := range tests {
+		client := NewRollingUpdaterClient(fakeClientFor("default", test.responses))
 		updater := RollingUpdater{
-			NewRollingUpdaterClient(fakeClientFor("default", test.responses)),
-			"default",
+			c:  client,
+			ns: "default",
+			scaleAndWait: func(rc *api.ReplicationController, retry *RetryParams, wait *RetryParams) (*api.ReplicationController, error) {
+				return client.GetReplicationController(rc.Namespace, rc.Name)
+			},
 		}
 		var buffer bytes.Buffer
+		var acceptor UpdateAcceptor
+		if test.accepted != nil {
+			acceptor = &testAcceptor{
```

Comment:
Not sure I follow what that adds since this code only cares about pass or fail from the acceptor, which I already exercise (via Accepted/Rejected booleans). Specific acceptor implementations elsewhere would have their own tests. Maybe I just misunderstood the scenario you're describing?